XenAdmin/Core/Updates.cs

```diff
@@ -503,32 +503,25 @@ private static bool PatchCanBeInstalledOnHost(XenServerPatch serverPatch, Host h
 
 
         /// <summary>
-        /// Returns the latest XenCenter version or null, if the current version is the latest. 
-        /// If a server version is provided, it returns the XenCenter version that is required to work with that server. 
-        /// If no server version is provided it will return the latestCr XenCenter.
+        /// Returns the minimum XenCenter version if it is greater than the current XC version,
+        /// or null, if the minimum XC version couldn't be found or the current XC version is enough
         /// </summary>
         /// <param name="serverVersion"></param>
         /// <returns></returns>
         public static XenCenterVersion GetRequiredXenCenterVersion(XenServerVersion serverVersion)
         {
-            if (XenCenterVersions.Count == 0)
+            var minXcVersion = serverVersion.MinimumXcVersion;

```

Comment:
These changes have removed the case of null serverVersion ("If no server version is provided it will return the latestCr XenCenter.") and iit will actually throw a null reference exception if serverVersion  is null, which could still be the case when used from XenCenterVersionCheck for the RPU wizard